qhana_plugin_registry/tasks/plugin_filter.py

```diff
@@ -12,21 +12,65 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+import json
 from celery.utils.log import get_task_logger
+from sqlalchemy.sql.expression import select
+
 from ..celery import CELERY
+from ..db.models.templates import TemplateTab
+from ..db.db import DB
+from ..db.models.plugins import RAMP
 
 _name = "qhana-plugin-registry.tasks.tabs"
 
 TASK_LOGGER = get_task_logger(_name)
 
 
+def evaluate_plugin_filter(filter_string: str) -> list[RAMP]:
+    """Get a list of plugins from a filter.
+
+    Args:
+        plugin_filter (str): The filter string.
+
+    Returns:
+        list: A list of plugins.
+    """
+    # TODO: store filter as dict in DB to avoid parsing
+    plugin_filter: dict = json.loads(filter_string)
+    plugins = DB.session.query(RAMP).all()
```

Comment:
This may grow large over time. Either implement filtering the plugins in batches (can be large batches ~500) or use other techniques to avoid loading too many data from DB at once. (Not high prio for this PR so just adding an issue for later + TODO comment here is sufficient for now.)

Suggestion:
Use a dict mapping plugin.id -> plugin. Then only use sets containing plugin IDs for the rest of this method (create a set from the dict keys for a set of all plugins). This allows you to remove the unsafe hash method from the plugin class and also avoids the equality test there entirely. The filters can then be entirely represented with set operations (at least `and` and `or`) and the leaf nodes create integer sets. (`not` may require special handling to get this correct...)

Note that set operations can also be used like this:
```python
# set difference
{1,2,3} - {1,2}  # {3}
# union
{1,2} | {2,3}  # {1,2,3}
```

Reply:
Batch filtering done in 506daa746689ae7790736a2ea9732c5333882933.